### ISSUE TITLE
Details List: When navigating from different route to a route where detail list fabric component is used, keyboard navigation does not focus on detail list rows

### ISSUE THREAD

**Issue by anubhavagg147** [OfficeDev/office-ui-fabric-react#10804]
### Describe the issue:

i am from microsoft flow team (https://preview.flow.microsoft.com/en-us/). We are using fabric component details list to show all the flow runs user had for a particular flow in flow properties page (https://preview.flow.microsoft.com/manage/environments/839eace6-59ab-4243-97ec-a5b8fcc104e4/flows/8d55f414-0332-4d14-a8af-e0e8a2479ea9/details) (you will not have access to this link as this is my flow, if you create a simple flow and run it then you will have  you list of runs, i will attach screenshot later to show how it reproes).
Now on using keyboard navigation, we are not able to access flow run lists row items. This only happens when we navigate from different route to flow properties page, if i refresh the flow properties page then i am able to access row items in detail list fabric component.

### Have you discussed this feature with our team, and if so, who
Kevin Coughlin (keco@microsoft.com)

###  Repro (Attachment)

![FabricBugRepro](https://user-images.githubusercontent.com/56557227/66770962-beb6ae80-ee6d-11e9-9521-8e4bfcf8b630.gif)

**Comment by marygans** [comment]
Hi @anubhavagg147 , thank you for submitting this issue!

@KevinTCoughlin I see you both have discussed this previously. Were you able to repro this outside of the context of the Flow? Is this work something you want to pick up? 

Thanks!

**Comment by KevinTCoughlin** [comment]
For context, we spoke offline prior to @anubhavagg147 opening this issue. What seemed to be occurring was that tab indices were not set on individual rows until they received focus. This would only occur during in-app navigation. The issue was not present when the page in question was loaded via hard refresh.

It seemed like there might be some on init focus issues at play that I am less familiar with.

**Comment by dzearing** [comment]
Is there any way we can reproduce it? 

@anubhavagg147  It seems that perhaps there is some sort of race where tabindex is not set anywhere within the new detailslist. I'd inspect the dom in the good and bad case and see if tabindex is -1 in one case and 0 in another.

**Comment by dzearing** [comment]
I figured out the bug!

When the page loads initially, CommandBar is the first component to mount which contains a FocusZone. On FZ mount, there is this code:

```tsx
      if (windowElement && _outerZones.size === 1) {
        this._disposables.push(on(windowElement, 'keydown', this._onKeyDownCapture, true));
      }
```

This code basically hooks up the first "outer" zone to the tabindex fixing code. This is only applied to the first zone on the page.

When you navigate, the command bar gets unmounted, and the event handler is unhooked prematurely. Instead, it should only be unhooked when the last outerZone is unmounted (size === 1).

The side effect here is that now the tabindexes will not be fixed on hitting tab.

The fix is: don't unhook the capture event until the last outer zone is unmounted. This is a relatively  serious bug and should be fixed immediately.
